pkg/controller/volume/persistentvolume/scheduler_binder_test.go

```diff
@@ -1380,22 +1410,64 @@ func TestBindPodVolumes(t *testing.T) {
 			},
 			shouldFail: true,
 		},
+		"node-affinity-fails": {
+			binding:    binding1aBound,
+			cachedPVs:  []*v1.PersistentVolume{pvNode1aBound},
+			cachedPVCs: []*v1.PersistentVolumeClaim{boundPVCNode1a},
+			nodes:      []*v1.Node{node1NoLabels},
+			shouldFail: true,
+		},
+		"node-affinity-fails-dynamic-provisioning": {
+			binding:         binding1aBoundSelectedNode,
```

Comment:
bindings is for static binding, and provisioned PVCs is for dynamic provisioning.  So this test case shouldn't need to set binding.

Reply:
The delay function takes in the pv/pvc from the binding, so it's currently needed. Or do you mean I should change its input? (It doesn't look [great](https://gist.github.com/bertinatto/5e6cf3ffa8c6c01b6634ef589262298f))

Reply:
What if the delay function takes in both binding and provisionedPVCs?

Reply:
Done. Had to re-organize the setup part because `BindPodVolumes` will [fail](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/scheduler_binder.go#L321-L323) if we pass in a `nil` binding (e.g., `nothing-to-bind-nil` test).